### PR TITLE
Few new features, small refactor, partially tested.

### PR DESCRIPTION
**Pull request by blaa**
My "take" on this project. I did similar stuff with bash scripts up to date, which wasn't pretty.

I like the ability to be able to connect USB freely when my screen is not locked. I'll further extend it to keep a list of currently connected devices and update it (instead of start devices).

Made better parsed config file and this allowed me to define 4 commands more. Feel free to hate it.

### CODE REVIEW

**Comment by hephaest0s** [comment]
Will look into it.

File: usbkill.py [hunk 1]
```diff
@@ -17,155 +18,291 @@
...
+    print(line)
+
+    if not log.path:
+        return
+    with open(log.path, 'a') as f:
+        # Empty line to separate log enties
+        f.write('\n')
+
+        # Log the message that needed to be logged:
+        f.write(line + '\n')
```
Comment (Brobin) [inline usbkill.py]: why not simply `f.write('\n{0}\n'.format(line))`

**Comment by Brobin** [comment]
I like the use of config parser. It's much more elegant that a bunch of janky string operations.

**Comment by pwnsdx** [comment]
+1 for the config parser ;)

**Comment by LinkseRat** [comment]
Maybe you can add an additional function (for raspberry) that formats the boot device so no forensics can find the 'sensitive' information? (Or mail it??)
I'm trying to configure my rspi2 with kali (not that simple) and this would be a nice tool when ....

**Comment by pwnsdx** [comment]
> Maybe you can add an additional function (for raspberry) that formats the boot device so no forensics can find the 'sensitive' information? (Or mail it??)

For Luks (cryptsetup): Adding a line to remove keys like `cryptsetup luksRemoveKey PARTITION` should do the trick, or if you can't, changing the password with a truely random string.
For FileVault (OS X): Any help is welcome

**Comment by blaa** [comment]
@LinkseRat  - that's why I added external kill script in my PR. It's generally advised to overwrite luks header with random bytes to make the drive look completely random + have a luksHeaderBackup stored somewhere safe. I'd use dd for this task, first determine size of the header by backing it up. It's certainly faster than remove/change key.

@Brobin  - yeah, of course it would be better. One of many places I think even. ;)

**Comment by hephaest0s** [comment]
Thanks, I added most of these features now. Could you take a look at this issue https://github.com/hephaest0s/usbkill/issues/51 ?

**Comment by blaa** [comment]
Thanks. Glad to be of help. Hope it can inspire you even more. You can still add support for different buses (and firewire is particularly known for DMA access to memory).

I was also going to add a filter - to allow only specified class of usb devices when the computer is `unlocked'. For example disallow new untrusted HID devices always (mouse wigglers), but allow for new unknown previously pendrives.

**Pull request closed without merging**